src/app/fdctl/run/tiles/fd_repair.c

```diff
@@ -126,15 +127,51 @@ mux_ctx( void * scratch ) {
 }
 
 static void
-send_packet( fd_repair_tile_ctx_t * ctx,
-             uint                   dst_ip_addr,
-             ushort                 dst_port,
-             uchar const *          payload,
-             ulong                  payload_sz,
-             ulong                  tsorig ) {
+send_intake_packet( fd_repair_tile_ctx_t * ctx,
+                    uint                   dst_ip_addr,
+                    ushort                 dst_port,
+                    uchar const *          payload,
+                    ulong                  payload_sz,
+                    ulong                  tsorig ) {
+  uchar * packet = fd_chunk_to_laddr( ctx->net_out_mem, ctx->net_out_chunk );
+
+  fd_memcpy( packet, ctx->intake_hdr, sizeof(fd_net_hdrs_t) );
+  fd_net_hdrs_t * hdr = (fd_net_hdrs_t *)packet;
+
+  hdr->udp->net_dport = dst_port;
+
+  memset( hdr->eth->dst, 0U, 6UL );
+  memcpy( hdr->ip4->daddr_c, &dst_ip_addr, 4UL );
+  hdr->ip4->net_id = fd_ushort_bswap( ctx->net_id++ );
+  hdr->ip4->check  = 0U;
+  hdr->ip4->net_tot_len  = fd_ushort_bswap( (ushort)(payload_sz + sizeof(fd_ip4_hdr_t)+sizeof(fd_udp_hdr_t)) );
+  hdr->ip4->check  = fd_ip4_hdr_check( ( fd_ip4_hdr_t const *)FD_ADDRESS_OF_PACKED_MEMBER( hdr->ip4 ) );
+
+  ulong packet_sz = payload_sz + sizeof(fd_net_hdrs_t);
+  fd_memcpy( packet+sizeof(fd_net_hdrs_t), payload, payload_sz );
+  hdr->udp->net_len   = fd_ushort_bswap( (ushort)(payload_sz + sizeof(fd_udp_hdr_t)) );
+  hdr->udp->check = fd_ip4_udp_check( *(uint *)FD_ADDRESS_OF_PACKED_MEMBER( hdr->ip4->saddr_c ),
+                                      *(uint *)FD_ADDRESS_OF_PACKED_MEMBER( hdr->ip4->daddr_c ),
+                                      (fd_udp_hdr_t const *)FD_ADDRESS_OF_PACKED_MEMBER( hdr->udp ),
+                                      packet + sizeof(fd_net_hdrs_t) );
+
+  ulong tspub = fd_frag_meta_ts_comp( fd_tickcount() );
+  ulong sig = fd_disco_netmux_sig( 0U, 0U, dst_ip_addr, DST_PROTO_OUTGOING, FD_NETMUX_SIG_MIN_HDR_SZ );
+  fd_mcache_publish( ctx->net_out_mcache, ctx->net_out_depth, ctx->net_out_seq, sig, ctx->net_out_chunk, packet_sz, 0UL, tsorig, tspub );
+  ctx->net_out_seq   = fd_seq_inc( ctx->net_out_seq, 1UL );
+  ctx->net_out_chunk = fd_dcache_compact_next( ctx->net_out_chunk, packet_sz, ctx->net_out_chunk0, ctx->net_out_wmark );
+}
```

Comment:
This is duplicated with the below function can you merge them insofar as that is possible?